pkg/controller/podautoscaler/horizontal.go

```diff
@@ -453,7 +453,12 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 		case desiredReplicas == 0:
 			//  never scale down to 0, reserved for disabling autoscaling
 			setCondition(hpa, autoscalingv2.ScalingLimited, v1.ConditionTrue, "TooFewReplicas", "the desired replica count was zero")
-			desiredReplicas = 1
+			// make sure we aren't below our minimum if one is defined
+			if hpa.Spec.MinReplicas != nil {
+				desiredReplicas = *hpa.Spec.MinReplicas
+			} else {
+				desiredReplicas = 1
+			}
```

Comment:
Instead of duplicating the condition I now aligned the order of cases to https://github.com/kubernetes/kubernetes/blob/e8eb858b2a28c12d2ffc669f70598ed83304e063/pkg/controller/podautoscaler/horizontal.go#L413-L418, which also prevents the broken behaviour. WDYT?